src/amp/lib/srcset-utils.ts

```diff
@@ -0,0 +1,41 @@
+import { bestFitImage } from '@root/src/amp/lib/image-fit';
+
+const containerWidths: number[] = [600, 800, 1000, 1200, 1600];
+// This distribution of numbers is a balance between providing enough pictures
+// to cover a wide variety of widths while keeping the set as small as possible.
+// Pascal, Feb 2020
+
+type DeduplicationPacket = {
+    items: SrcSetItem[];
+    widthsUsed: number[];
+};
+
+const removeDuplicateWidth = (items: SrcSetItem[]): SrcSetItem[] => {
+    const packet: DeduplicationPacket = items.reduce(
+        (acc: DeduplicationPacket, curr: SrcSetItem) => {
+            if (acc.widthsUsed.includes(curr.width)) {
+                return acc;
+            }
+            return {
+                items: acc.items.concat(curr),
+                widthsUsed: acc.widthsUsed.concat(curr.width),
+            };
+        },
+        {
+            items: [],
+            widthsUsed: [],
```

Comment:
Why not make this an object so testing for membership is more efficient?

Reply:
Yeah, I though of using a hash, but in this case, I thought this approach was slightly more natural and any performance penalty is negligible here :)

Reply:
( see http://jsben.ch/eF7Ru )